pandas/core/arrays/datetimes.py

```diff
@@ -1215,6 +1218,7 @@ def to_period(self, freq=None) -> PeriodArray:
 
             freq = res
 
+            freq = freq_to_period_freqstr(1, freq)
```

Comment:
wait this would break
```python
In [12]: pd.date_range('2020-01-01', periods=3, freq='2ME').to_period()
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
File offsets.pyx:4429, in pandas._libs.tslibs.offsets.to_offset()

ValueError: for Period, please use \'M\' instead of 'ME'

The above exception was the direct cause of the following exception:

ValueError                                Traceback (most recent call last)
Cell In[12], line 1
----> 1 pd.date_range('2020-01-01', periods=3, freq='2ME').to_period()

File ~/pandas-dev/pandas/core/indexes/extension.py:95, in _inherit_from_data.<locals>.method(self, *args, **kwargs)
     93 if "inplace" in kwargs:
     94     raise ValueError(f"cannot use inplace with {type(self).__name__}")
---> 95 result = attr(self._data, *args, **kwargs)
     96 if wrap:
     97     if isinstance(result, type(self._data)):

File ~/pandas-dev/pandas/core/arrays/datetimes.py:1222, in DatetimeArray.to_period(self, freq)
   1219     freq = res
   1221     freq = freq_to_period_freqstr(1, freq)
-> 1222 return PeriodArray._from_datetime64(self._ndarray, freq, tz=self.tz)

File ~/pandas-dev/pandas/core/arrays/period.py:318, in PeriodArray._from_datetime64(cls, data, freq, tz)
    316 if isinstance(freq, BaseOffset):
    317     freq = freq_to_period_freqstr(freq.n, freq.name)
--> 318 data, freq = dt64arr_to_periodarr(data, freq, tz)
    319 dtype = PeriodDtype(freq)
    320 return cls(data, dtype=dtype)

File ~/pandas-dev/pandas/core/arrays/period.py:1155, in dt64arr_to_periodarr(data, freq, tz)
   1152     data = data._values
   1154 reso = get_unit_from_dtype(data.dtype)
-> 1155 freq = Period._maybe_convert_freq(freq)
   1156 base = freq._period_dtype_code
   1157 return c_dt64arr_to_periodarr(data.view("i8"), base, tz, reso=reso), freq

File period.pyx:1768, in pandas._libs.tslibs.period._Period._maybe_convert_freq()

File offsets.pyx:4465, in pandas._libs.tslibs.offsets.to_offset()

ValueError: Invalid frequency: 2ME
```

Reply:
thanks, you are right. I'll fix it.